Verification failed when calling `exit()` from js shell

```
[alaska@fedora serenity]$ ./Meta/serenity.sh run lagom js
-- Configuring done (0.5s)
-- Generating done (0.3s)
-- Build files have been written to: /home/alaska/Downloads/serenity/Build/lagom
ninja: Entering directory `/home/alaska/Downloads/serenity/Build/lagom'
[0/2] Re-checking globbed directories...
[0/1] cd /home/alaska/Downloads/serenity/Build/lagom && /usr/bin/cmake -E env SERENITY_SOURCE_DIR=/home/alaska/Downloads/serenity /home/alaska/Downloads/serenity/Build/lagom/bin/js
> exit()
VERIFICATION FAILED: may_defer || !called_from_inside_function at /home/alaska/Downloads/serenity/Meta/Lagom/../../AK/Function.h:225
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-core.so.0(ak_verification_failed+0xef) [0x7fb83e1902df]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0(+0x30f393) [0x7fb83ef0f393]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::HeapBlock::deallocate(JS::Cell*) 0xa1) [0x7fb83ed683a1]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0(+0x163e61) [0x7fb83ed63e61]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Heap::sweep_dead_cells(bool, Core::ElapsedTimer const&) 0x16c) [0x7fb83ed62a4c]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Heap::~Heap() 0xba) [0x7fb83ed61eda]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::VM::~VM() 0x254) [0x7fb83f0a3214]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x40f1d6]
/lib64/libc.so.6(+0x401e6) [0x7fb83db811e6]
/lib64/libc.so.6(+0x4032e) [0x7fb83db8132e]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x407963]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x41111d]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0(+0x30f053) [0x7fb83ef0f053]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::NativeFunction::call() 0x25) [0x7fb83ef0eef5]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::NativeFunction::internal_call(JS::Value, JS::MarkedVector<JS::Value, 0ul>) 0x276) [0x7fb83ef0e9e6]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::call_impl(JS::VM&, JS::FunctionObject&, JS::Value, AK::Optional<JS::MarkedVector<JS::Value, 0ul> >) 0x22b) [0x7fb83edd36db]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::perform_call(JS::Bytecode::Interpreter&, JS::Value, JS::Bytecode::Op::CallType, JS::Value, JS::MarkedVector<JS::Value, 0ul>) 0x196) [0x7fb83ed300f6]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Op::Call::execute_impl(JS::Bytecode::Interpreter&) const 0x18d) [0x7fb83ed4711d]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run_bytecode() 0x28c) [0x7fb83ed3b57c]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run_and_return_frame(JS::Bytecode::Executable&, JS::Bytecode::BasicBlock const*, JS::Bytecode::CallFrame*) 0x24e) [0x7fb83ed3af3e]
/home/alaska/Downloads/serenity/Build/lagom/bin/../lib64/liblagom-js.so.0 JS::Bytecode::Interpreter::run(JS::Script&, JS::GCPtr<JS::Environment>) 0x37a) [0x7fb83ed3a5fa]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x40c36b]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x40b8b3]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x412d02]
/lib64/libc.so.6(+0x27b8a) [0x7fb83db68b8a]
/lib64/libc.so.6(__libc_start_main+0x8b) [0x7fb83db68c4b]
/home/alaska/Downloads/serenity/Build/lagom/bin/js() [0x406a25]
Illegal instruction
FAILED: CMakeFiles/run-lagom-target /home/alaska/Downloads/serenity/Build/lagom/CMakeFiles/run-lagom-target 
cd /home/alaska/Downloads/serenity/Build/lagom && /usr/bin/cmake -E env SERENITY_SOURCE_DIR=/home/alaska/Downloads/serenity /home/alaska/Downloads/serenity/Build/lagom/bin/js
ninja: build stopped: subcommand failed.
```

Curiously, calling other (undefined) functions work fine:
```
> foo()
Uncaught exception: 
[ReferenceError] 'foo' is not defined
 -> <unknown> (:0,0)
 -> <unknown> (:0,0)
 ```

--- a/Userland/Utilities/js.cpp
+++ b/Userland/Utilities/js.cpp
@@ -81,7 +81,8 @@ static bool s_disable_source_location_hints = false;
 static RefPtr<Line::Editor> s_editor;
 static String s_history_path = String {};
 static int s_repl_line_level = 0;
-static bool s_fail_repl = false;
+static bool s_keep_running_repl = true;
+static int s_exit_code = 0;
 
 static ErrorOr<void> print(JS::Value value, Stream& stream)
 {
@@ -124,7 +125,7 @@ static ErrorOr<String> read_next_piece()
         line_level_delta_for_next_line = 0;
 
         if (line_result.is_error()) {
-            s_fail_repl = true;
+            s_keep_running_repl = false;
             return String {};
         }
 
@@ -387,10 +388,11 @@ JS_DEFINE_NATIVE_FUNCTION(ReplObject::save_to_file)
 
 JS_DEFINE_NATIVE_FUNCTION(ReplObject::exit_interpreter)
 {
-    s_editor->save_history(s_history_path.to_deprecated_string());
-    if (!vm.argument_count())
-        exit(0);
-    exit(TRY(vm.argument(0).to_number(vm)).as_double());
+    if (vm.argument_count() != 0)
+        s_exit_code = TRY(vm.argument(0).to_number(vm)).as_double();
+
+    s_keep_running_repl = false;
+    return JS::js_undefined();
 }
 
 JS_DEFINE_NATIVE_FUNCTION(ReplObject::repl_help)
@@ -460,7 +462,7 @@ JS_DEFINE_NATIVE_FUNCTION(ScriptObject::print)
 
 static ErrorOr<void> repl(JS::Realm& realm)
 {
-    while (!s_fail_repl) {
+    while (s_keep_running_repl) {
         auto const piece = TRY(read_next_piece());
         if (Utf8View { piece }.trim(JS::whitespace_characters).is_empty())
             continue;
@@ -879,5 +881,5 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
             return 1;
     }
 
-    return 0;
+    return s_exit_code;
 }

